validator/client/service.go

```diff
@@ -15,44 +20,110 @@ type ValidatorService struct {
 	ctx       context.Context
 	cancel    context.CancelFunc
 	validator Validator
+	conn     *grpc.ClientConn
+	endpoint string
+	withCert string
 }
 
 // Config for the validator service.
 type Config struct {
-	BeaconClient    pb.BeaconServiceClient
-	ValidatorClient pb.ValidatorServiceClient
+	Endpoint string
+	CertFlag string
 }
 
 // NewValidatorService creates a new validator service for the service
 // registry.
 func NewValidatorService(ctx context.Context, cfg *Config) *ValidatorService {
 	ctx, cancel := context.WithCancel(ctx)
-	validator := &validator{
-		beaconClient:    cfg.BeaconClient,
-		validatorClient: cfg.ValidatorClient,
-	}
 	return &ValidatorService{
 		ctx:       ctx,
 		cancel:    cancel,
-		validator: validator,
+		endpoint: cfg.Endpoint,
+		withCert: cfg.CertFlag,
 	}
 }
 
 // Start the validator service. Launches the main go routine for the validator
 // client.
 func (v *ValidatorService) Start() {
+	var server grpc.DialOption
+	if v.withCert != "" {
+		creds, err := credentials.NewClientTLSFromFile(v.withCert, "")
+		if err != nil {
+			log.Errorf("Could not get valid credentials: %v", err)
+			return
+		}
+		server = grpc.WithTransportCredentials(creds)
+	} else {
+		server = grpc.WithInsecure()
+		log.Warn("You are using an insecure gRPC connection! Please provide a certificate and key to use a secure connection.")
+	}
+	providerURL, err := url.Parse(v.endpoint)
```

Comment:
Why do we have to do this? Can't we just provide the connection string to the grpc.Dial?